miner/worker.go

```diff
@@ -396,7 +396,7 @@ func (miner *Miner) commitBlobTransaction(env *environment, tx *types.Transactio
 	// isn't really a better place right now. The blob gas limit is checked at block validation time
 	// and not during execution. This means core.ApplyTransaction will not return an error if the
 	// tx has too many blobs. So we have to explicitly check it here.
-	if (env.blobs+len(sc.Blobs))*params.BlobTxBlobGasPerBlob > params.MaxBlobGasPerBlock {
+	if (env.blobs+len(tx.BlobHashes()))*params.BlobTxBlobGasPerBlob > params.MaxBlobGasPerBlock {
```

Comment:
I’m curious why this issue doesn’t occur in the upstream Geth implementation.

Reply:
It only happens when deriving, not sequencing. For L1 Geth, they don't derive, no never hit this case.

Reply:
Should we leave some comments here explaining the rationale behind this change so others can easily understand why it was made?

Reply:
Done in 5919b26987ee340280be050abb09afeb0f66e67e.